---
title: "A2"
author: "Annie Liu"
date: "2023-02-28"
output: 
  html_document:
      toc: true
      toc_depth: 3
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


### Dataset Introduction

In A1, I analyzed the dataset with the GEO ascension GSE75011, a dataset called Transcriptional profiling of TH2 cells identifies pathogenic features associated with asthma that was associated with the publication "Transcriptional Profilign of Th2 Cells Identifies Pathogenic Features Associated with Asthma" (Seumois et al. 2016). 

Asthma and allergic rhinitis are conditions affecting millions of people around the world, and these conditions are often comorbid with similar immunological hallmarks; however, not every person who has allergic rhinitis develop asthma in their lifetime. This study looks at the transcriptional profiling of Th2 cell enriched CD4 memory cells in patients with asthma, rhinitis, and healthy controls to attempt to gain genetic insights on these conditions. 

In A1, the aforementioned dataset was cleaned, filtered, and normalized using the edgeR protocol. The dataset contained gene names that were converted to HUGO symbols and then mapped to ensembl IDs. After data cleaning, 12973 genes were across all the samples were left with 12161 genes that had HUGO and ensmebl IDs.  


## Load Libraries

``` {r, message=FALSE, warning=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

if (!requireNamespace("edgeR", quietly = TRUE)) 
    BiocManager::install("edgeR")

if (!requireNamespace("ComplexHeatmap", quietly = TRUE))
    BiocManager::install("ComplexHeatmap")

if (!requireNamespace("circlize", quietly = TRUE))
    install.packages("circlize")

if (!requireNamespace("gprofiler2", quietly = TRUE))
    install.packages("gprofiler2")

library("BiocManager")
library("edgeR")
library(ComplexHeatmap)
library(circlize)
library("gprofiler2")

```


## Data Visualization
```{r, message=FALSE, warning=FALSE}
# Add normalized_count_data in
norm_data <- read.csv(file.path(getwd(), "data"))
norm_data = subset(norm_data, select= -X)
head(norm_data)

# Create numerical matrix that we can create a heatmap from

heatmap_matrix <- norm_data[, 4:ncol(norm_data)]
rownames(heatmap_matrix) <- norm_data$gene_hgnc
colnames(heatmap_matrix) <- colnames(norm_data[,4:ncol(heatmap_matrix)])

heatmap_matrix <- t(scale(t(heatmap_matrix)))

if(min(heatmap_matrix) == 0){
  heatmap_col = colorRamp2(c( 0, max(heatmap_matrix)), c("white", "red"))
} else {
  heatmap_col = colorRamp2(c(min(heatmap_matrix), 0, max(heatmap_matrix)), c("blue", "white", "red"))
}

current_heatmap <- Heatmap(as.matrix(heatmap_matrix),
                           show_row_dend = TRUE, show_column_dend= TRUE, col=heatmap_col, show_column_names = TRUE, show_row_names=FALSE, show_heatmap_legend=TRUE)

current_heatmap


# MDS plot 

cond <- data.frame(lapply(colnames(norm_data)[4:83], 
        FUN=function(x){unlist(strsplit(x, 
                        split = "_"))[c(1,2)]}))
colnames(cond) <- colnames(norm_data)[4:83]
rownames(cond) <- c("patient_id", "condition")
cond <- data.frame(t(cond))

expr_data <- as.matrix(norm_data[,4:83])
rownames(expr_data) <- norm_data$gene_hgnc
d = DGEList(counts=expr_data, group=cond$condition)
MDS_by_cond_edgeR <- plotMDS(d, labels=rownames(cond), col = c("darkgreen", "blue", "red")[factor(cond$condition)], main="MDS plot of all samples")

```

Going by what is present in the paper, we know that the sample groups are important in determining its value. Since we have two experimental groups here to one control, I am going to investigate each experimental group in relation to the other samples.

## Fitting model and computing differential expression
```{r, message=FALSE, warning=FALSE}
# model design 
conditions <- cond$condition
model_d <- model.matrix(~0 + conditions) 

head(model_d)

d <- estimateDisp(d, model_d)

# fit model 

fit <- glmQLFit(d, model_d)


# compare AS to all
contrast_AS <- makeContrasts(hctoall="conditionsAS-(conditionsHC + conditionsAS)/2", levels=model_d)
qlf_AS_to_all <- glmQLFTest(fit, contrast = contrast_AS)


# apply Benjamin-Hochberg 
tt_AS_to_all <- topTags(qlf_AS_to_all, adjust.method = "BH", n=nrow(d))
# 1934
length(which(tt_AS_to_all$table$PValue<0.05))
# 435
length(which(tt_AS_to_all$table$FDR<0.05))
output_AS_to_all <- merge(norm_data[,2:3], tt_AS_to_all, by.y=0, by.x=1, all.y=TRUE)
output_AS_to_all <- output_AS_to_all[order(output_AS_to_all$PValue),]
head(output_AS_to_all)

# compare AR to all
contrast_AR <- makeContrasts(hctoall="conditionsAR-(conditionsHC + conditionsAS)/2", levels=model_d)
qlf_AR_to_all <- glmQLFTest(fit, contrast = contrast_AR)

# apply Benjamin-Hochberg 
tt_AR_to_all <- topTags(qlf_AR_to_all, adjust.method = "BH", n=nrow(d))
# 1179
length(which(tt_AR_to_all$table$PValue<0.05))
# 17
length(which(tt_AR_to_all$table$FDR<0.05))
output_AR_to_all <- merge(norm_data[,2:3], tt_AR_to_all, by.y=0, by.x=1, all.y=TRUE)
output_AR_to_all <- output_AR_to_all[order(output_AR_to_all$PValue),]
head(output_AR_to_all)
```


## Volcano Plot of Differentially Expressed Genes
```{r, message=FALSE, warning=FALSE}
# Code adapted from the following https://www.nathalievialaneix.eu/doc/html/solution-edgeR-rnaseq.html

# Volcano Plot of AS to all 
volcano_Data_AS_to_All <- as.data.frame(cbind(tt_AS_to_all$table$logFC, -log10(tt_AS_to_all$table$FDR), tt_AS_to_all$table$PValue))
colnames(volcano_Data_AS_to_All) <- c("logFC", "negLogPval", "PValue")
volcano_Data_AS_to_All$color <- "black"
volcano_Data_AS_to_All$color[volcano_Data_AS_to_All$PValue < 0.05 & volcano_Data_AS_to_All$logFC > 0] <- "green"
volcano_Data_AS_to_All$color[volcano_Data_AS_to_All$PValue < 0.05 & volcano_Data_AS_to_All$logFC < 0] <- "orange"
plot(volcano_Data_AS_to_All$logFC,
     volcano_Data_AS_to_All$negLogPval,
     xlab="negLogPval",
     ylab= "logFC",
     col=volcano_Data_AS_to_All$color,
     pch=20,
     main="Volcano Plot of AS to all groups Differential Expression")
legend(x = "topright",    
       legend = c("Upregulated", "Downregulated", "Neither"),
       fill= c("green", "orange", "black")
       )
```


```{r, message=FALSE, warning=FALSE}
# Volcano Plot of AR to all
volcano_Data_AR_to_All <- as.data.frame(cbind(tt_AR_to_all$table$logFC, -log10(tt_AR_to_all$table$FDR), tt_AR_to_all$table$PValue))
colnames(volcano_Data_AR_to_All) <- c("logFC", "negLogPval", "PValue")
volcano_Data_AR_to_All$color <- "black"
volcano_Data_AR_to_All$color[volcano_Data_AR_to_All$PValue < 0.05 & volcano_Data_AR_to_All$logFC > 0] <- "green"
volcano_Data_AR_to_All$color[volcano_Data_AR_to_All$PValue < 0.05 & volcano_Data_AR_to_All$logFC < 0] <- "orange"
plot(volcano_Data_AR_to_All$logFC,
     volcano_Data_AR_to_All$negLogPval,
     xlab="negLogPval",
     ylab= "logFC",
     col=volcano_Data_AR_to_All$color,
     pch=20,
     main="Volcano Plot of AR to all groups Differential Expression")
legend(x = "topright",          # Position
       legend = c("Upregulated", "Downregulated", "Neither"),
       fill= c("green", "orange", "black")
       )
```

## Heatmap of Top Hits
```{r, message=FALSE, warning=FALSE}
# normalized counts

normalized_c <- log2(cpm(d) + 1)
heatmap_matrix_diff <- normalized_c


ha_colours <- c("darkgreen","blue","red")
names(ha_colours) <- unique(cond$condition)
ha <- HeatmapAnnotation(df=data.frame(
  type = cond$condition),
  col =  list(type = ha_colours))

AS_top_hits <- rownames(tt_AS_to_all)[which(tt_AS_to_all$table$FDR < 0.05)]
AS_heatmap_matrix_tophits <- t(scale(t(heatmap_matrix_diff[which(rownames(heatmap_matrix_diff)  %in% AS_top_hits),])))

if(min(AS_heatmap_matrix_tophits) == 0){
  AS_heatmap_col = colorRamp2(c(0, max(AS_heatmap_matrix_tophits)),
                              c("white", "red"))
} else {
  AS_heatmap_col = colorRamp2(c(min(heatmap_matrix), 0, max(heatmap_matrix)), c("blue", "white", "red"))
}


AS_heatmap <- Heatmap(as.matrix(AS_heatmap_matrix_tophits),
                      cluster_rows = TRUE, show_row_dend = TRUE, 
                      cluster_columns = TRUE, show_column_dend = FALSE,
                      col=AS_heatmap_col, show_column_names = FALSE,
                      show_row_names = FALSE, show_heatmap_legend=TRUE, top_annotation = ha)


AS_heatmap
```


```{r, message=FALSE, warning=FALSE}
AR_top_hits <- rownames(tt_AR_to_all)[which(tt_AR_to_all$table$FDR < 0.05)]
AR_heatmap_matrix_tophits <- t(scale(t(heatmap_matrix_diff[which(rownames(heatmap_matrix_diff)  %in% AR_top_hits),])))

if(min(AS_heatmap_matrix_tophits) == 0){
  AR_heatmap_col = colorRamp2(c(0, max(AR_heatmap_matrix_tophits)),
                              c("white", "red"))
} else {
  AR_heatmap_col = colorRamp2(c(min(heatmap_matrix), 0, max(heatmap_matrix)), c("blue", "white", "red"))
}


AR_heatmap <- Heatmap(as.matrix(AR_heatmap_matrix_tophits),
                      cluster_rows = TRUE, show_row_dend = TRUE, 
                      cluster_columns = TRUE, show_column_dend = FALSE,
                      col=AS_heatmap_col, show_column_names = FALSE,
                      show_row_names = FALSE, show_heatmap_legend=TRUE, top_annotation = ha)


AR_heatmap

```

## Differential Gene Expression Summary
**Calculate p-values for each of the genes in your expression set. How many genes were significantly differentially expressed? What thresholds did you use and why?**

In our comparison of AS to all, there were 1934 genes with p-values less than 0.05. In our comparison of AR to all, there were 1179 genes with p-values less than 0.05.

**Multiple hypothesis testing - correct your p-values using a multiple hypothesis correction method. Which method did you use? And Why? How many genes passed correction?**

I used Benjamin-Hochberg as a multiple hypothesis correction method because this is what the original publishers of the dataset used. In our comparison of AS to all, 435 genes that passed correction. In our comparison of AR to all, 17 genes passed correction. 

**Show the amount of differentially expressed genes using an MA Plot or a Volcano plot. Highlight genes of interest.**

See the section "Volcano Plot of Differentially Expressed Genes"

**Visualize your top hits using a heatmap. Do you conditions cluster together? Explain why or why not.**

See the section "Heatmap of Top Hits" for the graphs. What is striking to me is that in both the AR to All comparison and AS to All comparison, the HC (control) group clearly clusters together which implies that the differential expression in the AR and AS group is more similar to each other than to the HC group. This supports the notion that AS and AR have similar pathologies as stated in the original paper. 


# Create threshold list of genes 
```{r, message=FALSE, warning=FALSE}
# AS to all differentially expressed genes
# number of genes upregulated 
length(which(output_AS_to_all$PValue < 0.05 & output_AS_to_all$logFC >0))

# number of genes downregulated
length(which(output_AS_to_all$PValue < 0.05 & output_AS_to_all$logFC < 0))

output_AS_to_all[, "rank"] <- -log10(output_AS_to_all$PValue) * sign(output_AS_to_all$logFC)
output_AS_to_all <- output_AS_to_all[order(output_AS_to_all$rank),]
upregulated_AS <- output_AS_to_all$gene_hgnc[which(output_AS_to_all$PValue < 0.05 & output_AS_to_all$logFC > 0)]
downregulated_AS <- output_AS_to_all$gene_hgnc[which(output_AS_to_all$PValue < 0.05 & output_AS_to_all$logFC < 0)]


# AR to all differentially expressed genes
# number of genes upregulated 
length(which(output_AR_to_all$PValue < 0.05 & output_AR_to_all$logFC >0))

# number of genes downregulated
length(which(output_AR_to_all$PValue < 0.05 & output_AR_to_all$logFC < 0))

output_AR_to_all[, "rank"] <- -log10(output_AR_to_all$PValue) * sign(output_AR_to_all$logFC)
output_AR_to_all <- output_AR_to_all[order(output_AR_to_all$rank),]
upregulated_AR <- output_AR_to_all$gene_hgnc[which(output_AR_to_all$PValue < 0.05 & output_AR_to_all$logFC > 0)]
downregulated_AR <- output_AR_to_all$gene_hgnc[which(output_AR_to_all$PValue < 0.05 & output_AR_to_all$logFC < 0)]

```

gprofiler2 has a intuitive and easy to use R package. Given that we've already had experience using it from our homework assignment and that we currently have a list of genes, it's the most accessible tool for us to use for this assignment.

## Using gprofiler2 to conduct overexpression analysis 
```{r, message=FALSE, warning=FALSE}
# code adapted from the gprofiler paper (Kolberg et al. 2020)

# using whole differentially expressed gene list 

gprof_AS_all = gost(output_AS_to_all$gene_hgnc[which(output_AS_to_all$PValue < 0.05)], ordered_query=TRUE, correction_method = "fdr", sources= c("GO:BP", "WP"))

nrow(gprof_AS_all$result) # 2970

gprof_AR_all =  gost(output_AR_to_all$gene_hgnc[which(output_AR_to_all$PValue < 0.05)], ordered_query=TRUE, correction_method = "fdr", sources= c("GO:BP", "WP"))

nrow(gprof_AR_all$result) # 1489

# upregulated, downregulated genes 

gprof_AS_up = gost(upregulated_AS, organism= "hsapiens", ordered_query=TRUE, correction_method = "fdr", sources= c("GO:BP", "WP"))

nrow(gprof_AS_up$result) # 595


gprof_AS_down = gost(downregulated_AS, organism= "hsapiens", ordered_query=TRUE, correction_method = "fdr", sources= c("GO:BP", "WP"))

nrow(gprof_AS_down$result) # 2644

gprof_AR_up = gost(upregulated_AR, organism= "hsapiens", ordered_query=TRUE, correction_method = "fdr", sources= c("GO:BP", "WP"))


nrow(gprof_AR_up$result) # 465

gprof_AR_down = gost(downregulated_AR, organism= "hsapiens", ordered_query=TRUE, correction_method = "fdr", sources= c("GO:BP", "WP"))

nrow(gprof_AR_down$result) # 1050
```

gprofiler has a plotting function that we can use the better visualize these results. 

```{r, fig.cap="ORA Results for all Differentially expressed genes in the AS to all group"}
gostplot(gprof_AS_all, interactive = FALSE)
```


```{r, fig.cap="ORA Results for all Differentially expressed genes in the AR to all group"}
gostplot(gprof_AR_all, interactive = FALSE)
```


```{r, fig.cap="ORA Results for upregulated Differentially expressed genes in the AS to all group"}
gostplot(gprof_AS_up, interactive = FALSE)
```


```{r, fig.cap="ORA Results for downregulated Differentially expressed genes in the AS to all group"}
gostplot(gprof_AS_down, interactive = FALSE)
```


```{r, fig.cap="ORA Results for upregulated Differentially expressed genes in the AR to all group"}
gostplot(gprof_AR_up, interactive = FALSE)
```

```{r, fig.cap="ORA Results for downregulated Differentially expressed genes in the AR to all group"}
gostplot(gprof_AR_down, interactive = FALSE)
```

## Thresholded over-representation analysis
**Which method did you choose and why?**

I chose gprofiler because I found its interface intuitive and easy to use in our homework assignment. In addition, gprofiler also updates its data regularly and has a very straightforward R package. Both of these factors made it a no brainer for this assignment. 

**What annotation data did you use and why? What version of the annotation are you using?**

I used GO:BP for gene ontology as this data source is updated very regularly. I used Wikipathways as I was interested in biological pathway analysis, and I like that Wikipathways is an open sourced project with a lot of available data that is not vetted by a certain group (Bohler et al. 2016). 


```{r, message=FALSE, warning=FALSE}
# get annotation data versions

gprof_version <- get_version_info(organism="hsapiens")

GO_BP_ver <- gprof_version$sources$`GO:BP`$version

GO_BP_ver # "annotations: BioMart\nclasses: releases/2022-12-04"

WP_ver <- gprof_version$sources$WP$version

WP_ver # "20221210"

```


**How many genesets were returned with what thresholds?**

We used a P value of <0.05 as is convention. The genesets returned are as follows: 

```{r, message=FALSE, warning=FALSE}
# get number of gene sets 
nrow(gprof_AS_all$result) # 2970
nrow(gprof_AR_all$result) # 1489

nrow(gprof_AS_up$result) # 595
nrow(gprof_AS_down$result) # 2644

nrow(gprof_AR_up$result) # 465
nrow(gprof_AR_down$result) # 1050

```
2970 genes were returned in our full list AS to all comparison.
1489 genes were returned in our full list AR to all comparison.
595 genes were returned in our upregulated list AS to all comparison.
2644 genes were returned in our downregulated list AS to all comparison.
465 genes were returned in our upregulated AR to all comparison.
1050 genes were returned in our downregulated AR to all comparison.


**Run the analysis using the up-regulated set of genes, and the down-regulated set of genes separately. How do these results compare to using the whole list (i.e all differentially expressed genes together vs. the up-regulated and down regulated differentially expressed genes separately)?**

We will look at the GEO:BP results as every list had hits in GEO:BP but not all had hits in WP.


```{r, message=FALSE, warning=FALSE}
# AS to all
gprof_results_AS_all_diff <- gprof_AS_all$result$term_name[1:5]
gprof_results_AS_up_diff <- gprof_AS_up$result$term_name[1:5]
gprof_results_AS_down_diff <- gprof_AS_down$result$term_name[1:5]

ora_AS_comparison <- data.frame(gprof_results_AS_all_diff, 
                                gprof_results_AS_up_diff,
                                gprof_results_AS_down_diff)

colnames(ora_AS_comparison) <- c("Whole List", "Upregulated", "Downregulated")

ora_AS_comparison
```


Since in the AS to other groups comparison, there were no significant results in the upregulated list, but quite a number of significant results in the downregulated list, thus the whole list results reflect this distinction.

```{r, message=FALSE, warning=FALSE}
# AR to all

gprof_results_AR_all_diff <- gprof_AR_all$result$term_name[1:5]
gprof_results_AR_up_diff <- gprof_AR_up$result$term_name[1:5]
gprof_results_AR_down_diff <- gprof_AR_down$result$term_name[1:5]

ora_AR_comparison <- data.frame(gprof_results_AR_all_diff, 
                                gprof_results_AR_up_diff,
                                gprof_results_AR_down_diff)

colnames(ora_AR_comparison) <- c("Whole List", "Upregulated", "Downregulated")

ora_AR_comparison
```

In contrast to the AS to all group, the AR to all group did not have signficant hits in the upregulated and downregulated graphs as shown above. Thus, the term results for the query including the entire gene list is not identical to either the upregulated gene list result or downregulated gene list result.


## Interpretation
**Do the over-representation results support conclusions or mechanism discussed in the original paper?**

The over-representation results differ from the conclusions and mechanisms discussed in the original paper. 
The original paper highlighted Il17RB, MAN1A1, CASP2 among other genes of interest whole the top genes in my results include HMGB2, ODC1, H1-10 and so forth. 

It is relevant to note that the original paper used weighted gene coexpression network analysis which is a method of finding correlated genes which would have likely contributed differing gene lists. They also used the DESeq2 package for their analysis while I used edgeR and these packages are based on similar but different hypotheses.


**Can you find evidence, i.e. publications, to support some of the results that you see. How does this evidence support your results.**

The top gene that this analysis found was HMGB2 which, according to its Uniprot profile (https://www.uniprot.org/uniprotkb/P26583/entry), is a gene
involved in immune responses; however, there does not appear to be a publication currently linking it to asthma or allergic rhinitis.


## References

Bohler, A., Wu, G., Kutmon, M., Pradhana, L. A., Coort, S. L., Hanspers, K., Haw, R., Pico, A. R., & Evelo, C. T. (2016). Reactome from a WikiPathways Perspective. PLoS computational biology, 12(5), e1004941. https://doi.org/10.1371/journal.pcbi.1004941

Isserlin, R. (2023). Lecture 6 - Differential Expression. Quercus.
https://q.utoronto.ca/courses/294979/files/24035541?module_item_id=4287845

Isserlin, R. (2023). Lecture 7 -Annotation Dataset and Intro to Pathway analysis. Quercus. https://q.utoronto.ca/courses/294979/files/24035553?module_item_id=4287856

Kolberg, L., Raudvere, U., Kuzmin, I., Vilo, J., & Peterson, H. (2020). gprofiler2 -- an R package for gene list functional enrichment analysis and namespace conversion toolset g:Profiler. F1000Research, 9, ELIXIR-709. https://doi.org/10.12688/f1000research.24956.2

Langfelder, P., & Horvath, S. (2008). WGCNA: an R package for weighted correlation network analysis. BMC bioinformatics, 9, 559. https://doi.org/10.1186/1471-2105-9-559

Gu, Z., Gu, L., Eils, R., Schlesner, M., & Brors, B. (2014). circlize Implements and enhances circular visualization in R. Bioinformatics (Oxford, England), 30(19), 2811–2812. https://doi.org/10.1093/bioinformatics/btu393

Moisan, A., Gonzales, I., & Villa-Vialaneix, N. (2014, December 10). Practical statistical analysis of RNA-Seq data - edgeR. Retrieved March 14, 2023, from Nathalievialaneix.eu website: https://www.nathalievialaneix.eu/doc/html/solution-edgeR-rnaseq.html

Raudvere, U., Kolberg, L., Kuzmin, I., Arak, T., Adler, P., Peterson, H. and Vilo, J., 2019. g: Profiler: a web server for functional enrichment analysis and conversions of gene lists (2019 update). Nucleic Acids Research, 47(W1), pp.W191-W198.; doi:10.1093/nar/gkz369

Robinson, M. D., McCarthy, D. J., & Smyth, G. K. (2010). edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics (Oxford, England), 26(1), 139–140. https://doi.org/10.1093/bioinformatics/btp616

Seumois, G., Zapardiel-Gonzalo, J., White, B., Singh, D., Schulten, V., Dillon, M., Hinz, D., Broide, D. H., Sette, A., Peters, B., & Vijayanand, P. (2016). Transcriptional Profiling of Th2 Cells Identifies Pathogenic Features Associated with Asthma. Journal of immunology (Baltimore, Md. : 1950), 197(2), 655–664. https://doi.org/10.4049/jimmunol.1600397

Zuguang Gu, Roland Eils, Matthias Schlesner, Complex heatmaps reveal patterns and correlations in multidimensional genomic data, Bioinformatics, Volume 32, Issue 18, September 2016, Pages 2847–2849, https://doi.org/10.1093/bioinformatics/btw313


